src/os_maild/os_maild_client.c

```diff
@@ -208,14 +208,16 @@ MailMsg *OS_RecvMailQ(file_queue *fileq, struct tm *p, MailConfig *Mail, MailMsg
              extra_data,
              logs);
 #else
-    snprintf(mail->body, BODY_SIZE - 1, MAIL_BODY,
+    if (snprintf(mail->body, BODY_SIZE - 1, MAIL_BODY,
              al_data->date,
              al_data->location,
              al_data->rule,
              al_data->level,
              al_data->comment,
              extra_data,
-             logs);
+             logs) >= OS_MAXSTR + OS_SIZE_1024 -1) {
```

Comment:
Why are we comparing `OS_MAXSTR + 1023` instead of `BODY_SIZE - 1`?

Reply:
Yes, you are right. It is better to use `BODY_SIZE -1`. I used `OS_MAXSTR + 1023` because `BODY_SIZE  = OS_MAXSTR + OS_SIZE_1024` but it is not necessary.